src/Compilers/Core/Portable/MetadataReader/MetadataTypeName.cs

```diff
@@ -30,16 +32,30 @@ internal partial struct MetadataTypeName
         /// </summary>
         private string _namespaceName;
 
+        /// <summary>
+        /// <see cref="ReadOnlyMemory{T}"/> version of <see cref="_namespaceName"/>.  Preferred when possible to avoid
+        /// the copy of the portion of <see cref="_fullName"/> used for <see cref="_namespaceName"/>.
+        /// </summary>
+        private ReadOnlyMemory<char> _namespaceNameMemory;
```

Comment:
we could consider making this type *ONLY* expose data with ROMs (and not strings).  But the change then becomes much more viral as all consumers must be updated to support that.

Note that *only* exposing ROM can be worse as well *if* the consumers then ToString those ROMs (as each ToString gets a fresh copy, while this type caches the request the first time).